src/vs/workbench/contrib/terminal/browser/xterm/markNavigationAddon.ts

```diff
@@ -349,6 +321,13 @@ export class CommandNavigationAddon extends Disposable implements ICommandTracke
 		this._scrollToMarker(this._currentMarker, scrollPosition);
 	}
 
+	scrollToClosestMark(startMarker: IMarker, endMarker?: IMarker | undefined, highlight?: boolean | undefined): void {
```

Comment:
unknowns/ to consider:
- the name of this function
- is this the expected behavior?
- should it be configurable via some properties passed in ?
- when `endMarker` isn't provided but `highlight` is, should it select `startMarker.line`?
- is `endMarker` only important when `highlight` is true or should it scroll to the middle of start and end line?

Reply:
For highlight I wanted this to show the focus color bar on the left, rather than selecting the actual text. So highlighting just the start marker would look like:

![image](https://user-images.githubusercontent.com/2193314/185000040-15c7c07d-1c2a-4e1b-a6e8-bcca9e8106de.png)

And highlighting start to end:

![image](https://user-images.githubusercontent.com/2193314/185000078-5cf0f52c-40fb-4dad-9cb2-dc941e317a9f.png)

For the latter you need to consider the lifecycle of the decorations. I think creating a separate decoration for every line should be fine since this will typically not be used often and when it is the number of decorations is capped to the number of lines in the viewport.

Reply:
> should it be configurable via some properties passed in ?

What did you have in mind? It would be great to avoid a setting if we can get the behavior most people want

Reply:
we have the height property - so not sure we need decorations for each line?